### PR TITLE
Electron app

### PR DESCRIPTION
**Pull request by EFHIII**
Reorganized files and added Electron App

### CODE REVIEW

File: src/client/articleQueue.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+function branch(steps, article, parent, searched, searchedBranch, queued) {
+  let parentID = parent ? getArticleID(parent) : false;
+  let articleID = getArticleID(article);
+  if(parentID && searched.hasOwnProperty(parentID + '->' + articleID)) {
+    return;
+  }
+  if(parent) {
+    searched[parentID + '->' + articleID] = true;
+    parent.relevance += -1 / Math.pow(2, Math.log(article.numCitations / 1000 + 1) / Math.log(10)) + 1;
```
Comment (aditink) [inline src/client/articleQueue.js:69]: Could this expression -1 / Math.pow(2, Math.log(article.numCitations / 1000 + 1) / Math.log(10)) be extracted to a function?

File: src/electron/src/main.js [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+    page += 10;
+    children = children.concat(newChildren); //concat children to output array
+
+    for(let i = 0; i < newChildren.length; i++) {
+      lowest = Math.min(lowest, newChildren[i].numCitations);
+    }
+    if(newChildren.length <= 0) {
+      lowest = 0;
+    }
+    //console.log(children.length + '/' + MaximumArticles); //progress over maximum articles
```
Comment (aditink) [inline src/electron/src/main.js:114]: Nit: remove/put under debug flag

File: src/client/articleQueue.js (hunk 1 above)
Comment (EFHIII) [reply src/client/articleQueue.js:69]: Yea,
This is one of the parts I've been playing with to see what gives better results.

**Comment by EFHIII** [comment]
I'm still ironing out a few things with the Electron app, I probably won't be able to get to refactoring the code until Monday or so.

File: src/client/articleQueue.js (hunk 1 above)
Comment (EFHIII) [reply src/client/articleQueue.js:69]:
```js
parent.relevance += (-1 / Math.pow(2, Math.log(article.numCitations / 1000 + 1) / Math.log(10)) + 1) * Math.pow(1 - IndexWeight, index + 5) / 100 * parentWeight;
```
Was the most recent version, but I'm actually having it commented out at the moment, which seems to work well. (line 196 in main.js of the Electron app)

**Pull request merged**